lib/ohai/plugins/shard.rb

```diff
@@ -0,0 +1,43 @@
+#
+# Author:: Phil Dibowitz <phil@ipom.com>
+# Copyright:: Copyright (c) 2016 Facebook, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+def get_dmi_thing(dmi, thing)
+  %w{system base_board chassis}.each do |section|
+    unless dmi[:thing][thing].compact.empty?
+      return dmi[:thing][thing]
+    end
+  end
+end
+
+Ohai.plugin(:ShardSeed) do
+  depends "hostname", "dmi"
+  provides "shard_seed"
+
+  collect_data(:default) do
+    base = Ohai.config[:plugin][:shard_seed][:base] || :hostname
+    data = case base
+           when :hostname
+             fqdn
+           when :serial
+             get_dmi_thing(dmi, :serial_number)
+           when :uuid
+             get_dmi_thing(dmi, :uuid)
+           end
+    shard_seed Digest::MD5.hexdigest(data)[0...7].to_i(16)
```

Comment:
Anyone know if this will be an issue with FIPS? ISTR that the md5 functions get removed if you have OpenSSL in FIPS mode (groan). Alternatively, just using SHA-2 probably won't hurt anything.